backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/directory/impl/LocalHeapDirectoryHolder.java

```diff
@@ -26,7 +26,9 @@ final class LocalHeapDirectoryHolder implements DirectoryHolder {
 
 	@Override
 	public void start() {
-		this.directory = new ByteBuffersDirectory( lockFactory );
+		if ( directory == null ) {
+			directory = new ByteBuffersDirectory( lockFactory );
+		}
```

Comment:
if you revert it, you will see that `LuceneIndexRestartFromPreviousIntegrationIT` won't pass anymore.
The check allows us to not create a new Lucene `Directory` when the holder is reused.
I think that actually I should do the same for the `LocalFileSystemDirectoryHolder`.

Reply:
alternatively we could make the index accessor not call the holder start when the holder is reused, maybe it is cleaner and it would avoid to change the directory holder implementations. Going to use this latter option